okio/src/jvmTest/kotlin/okio/PipeKotlinTest.kt

```diff
@@ -609,6 +609,157 @@ class PipeKotlinTest {
     assertTrue(foldedSinkClosed)
   }
 
+  @Test fun cancelPreventsSinkWrite() {
+    val pipe = Pipe(8)
+    pipe.cancel()
+
+    val pipeSink = pipe.sink.buffer()
+    pipeSink.writeUtf8("hello world")
+
+    try {
+      pipeSink.emit()
```

Comment:
Are these tests missing `fail()` in order to ensure the catch block is run?

Reply:
@swankjesse even better, use `assertThrows`.

Reply:
Nah, they’re missing `fail()` because I’m clumsy. Fixed!

Reply:
Can’t use `assertThrows` because we don’t depend on kotlin.test. Should do that in a follow-up change!